obspy/io/shapefile/core.py

```diff
@@ -0,0 +1,282 @@
+
+            # setting fields with `None` results in values of `0.000`
+            # need to really omit setting values if they are `None`
+            if net.code is not None:
+                feature.SetField(native_str("Network"),
+                                 native_str(net.code))
+            if sta.code is not None:
+                feature.SetField(native_str("Station"),
+                                 native_str(sta.code))
+            if sta.latitude is not None:
+                feature.SetField(native_str("Latitude"), sta.latitude)
+            if sta.longitude is not None:
+                feature.SetField(native_str("Longitude"), sta.longitude)
+            if sta.elevation is not None:
+                feature.SetField(native_str("Elevation"), sta.elevation)
+            if sta.start_date is not None:
+                date = sta.start_date
+                # ESRI shapefile attributes are stored in dbf files, which can
+                # not store datetimes, only dates. We still need to use the
+                # GDAL API with precision up to seconds (aiming at other output
+                # drivers of GDAL; `100` stands for GMT)
+                feature.SetField(native_str("StartDate"), date.year,
+                                 date.month, date.day, date.hour, date.minute,
+                                 date.second, 100)
+            if sta.end_date is not None:
+                date = sta.end_date
+                # ESRI shapefile attributes are stored in dbf files, which can
+                # not store datetimes, only dates. We still need to use the
+                # GDAL API with precision up to seconds (aiming at other output
+                # drivers of GDAL; `100` stands for GMT)
+                feature.SetField(native_str("StartDate"), date.year,
+                                 date.month, date.day, date.hour, date.minute,
+                                 date.second, 100)
+            if channel_list:
+                feature.SetField(native_str("Channels"),
+                                 native_str(channel_list))
+
+            if sta.latitude is not None and sta.longitude is not None:
+                point = ogr.Geometry(ogr.wkbPoint)
+                point.AddPoint(sta.longitude, sta.latitude)
+                feature.SetGeometry(point)
+
+        layer.CreateFeature(feature)
+        # Destroy the feature to free resources
+        feature.Destroy()
+
+    # Destroy the data source to free resources
+    data_source.Destroy()
+
+
+def _get_WGS84_spatial_reference():
+    # create the spatial reference
+    sr = osr.SpatialReference()
+    # Simpler and feels cleaner to initialize by EPSG code but that depends on
+    # a csv file shipping with GDAL and some GDAL environment paths being set
+    # correctly which was not the case out of the box in anaconda, so better
+    # hardcode this bit.
+    # sr.ImportFromEPSG(4326)
+    wgs84_wkt = \
+        """
```

Comment:
Just never know how to make long strings PEP8 compliant and this way didn't spring any alarm signs for me, the parser for this string doesn't seem to care about any spaces in between and neither do I. ;)